homeassistant/components/tradfri/sensor.py

```diff
@@ -67,3 +81,46 @@ def __init__(
     def _refresh(self) -> None:
         """Refresh the device."""
         self._attr_native_value = self.coordinator.data.device_info.battery_level
+
+
+class AirQualitySensor(TradfriBaseEntity, SensorEntity):
+    """
+    The platform class required by Home Assistant.
+
+    Following the discussion here, the sensor seem to be reporting in microgram per
+    m3:
+    https://github.com/dresden-elektronik/deconz-rest-plugin/
+    issues/5351#issuecomment-968362604
+    """
+
+    _attr_device_class = SensorDeviceClass.AQI
+    _attr_native_unit_of_measurement = CONCENTRATION_MICROGRAMS_PER_CUBIC_METER
+
+    def __init__(
+        self,
+        device_coordinator: TradfriDeviceDataUpdateCoordinator,
+        api: Callable[[Command | list[Command]], Any],
+        gateway_id: str,
+    ) -> None:
+        """Initialize a air quality sensor."""
+        super().__init__(
+            device_coordinator=device_coordinator,
+            api=api,
+            gateway_id=gateway_id,
+        )
+
+        self._refresh()  # Set initial state
+
+        self._attr_unique_id = f"{self._attr_unique_id}-air_quality"
+
+    def _refresh(self) -> None:
+        """Refresh the device."""
+        aqi_value = self.coordinator.data.air_purifier_control.air_purifiers[
+            0
+        ].air_quality
+
+        # The sensor has this value if the device is turned off
+        if aqi_value == 65535:
+            aqi_value = None
+
+        self._attr_native_value = aqi_value
```

Comment:
Would it be cleaner to move this function into the coordinator itself so you could just call

`self._attr_native_value = coordinator.get_aqi_value` in your init block?

Reply:
Thank you for your review! I’ve been working with @MartinHjelmare on the implementation of the coordinator for the past few weeks. I feel he’s got a clean view on the path forward, so I’ll follow his advice on this one.